examples/step-76/doc/intro.dox

```diff
@@ -0,0 +1,468 @@
+for all faces of the cells.
+
+Currently, cell-centric loops in deal.II only work for uniformly refined meshes.
+
+
+<h3>Providing lambdas to MatrixFree loops</h3>
+
+The examples given above have already used lambdas, which have been provided to
+matrix-free loops. The following short examples present how to transform functions between
+a version where a class and a pointer to one of its methods are used and a
+variant where lambdas are utilized.
+
+In the following code, a class and a pointer to one of its methods, which should
+be interpreted as cell integral, are passed to MatrixFree::loop():
+
+@code
+void
+local_apply_cell(const MatrixFree<dim, Number> &              data,
+                 VectorType &                                 dst,
+                 const VectorType &                           src,
+                 const std::pair<unsigned int, unsigned int> &range) const
+{
+  FEEvaluation<dim, degree, degree + 1, 1, Number> phi(data);
+  for (unsigned int cell = range.first; cell < range.second; ++cell)
+    {
+      phi.reinit(cell);
+      phi.gather_evaluate(src, cell_evaluation_flags);
+
+      // some operations on the quadrature points
+
+      phi.integrate_scatter(cell_evaluation_flags, dst);
+    }
+}
+@endcode
+
+@code
+matrix_free.cell_loop(&Operator::local_apply_cell, this, dst, src);
+@endcode
+
+However, it is also possible to pass an anonymous function via a lambda function
+with the same result:
+
+@code
+matrix_free.template cell_loop<VectorType, VectorType>(
+  [&](const auto &data, auto &dst, const auto &src, const auto range) {
+    FEEvaluation<dim, degree, degree + 1, 1, Number> phi(data);
+    for (unsigned int cell = range.first; cell < range.second; ++cell)
+      {
+        phi.reinit(cell);
+        phi.gather_evaluate(src, cell_evaluation_flags);
+
+        // some operations on the quadrature points
+
+        phi.integrate_scatter(cell_evaluation_flags, dst);
+      }
+  },
+  dst,
+  src);
+@endcode
+
+<h3>VectorizedArrayType</h3>
+
+The class VectorizedArray<Number> is a key component to achieve the high
+node-level performance of the matrix-free algorithms in deal.II.
+It is a wrapper class around a short vector of $n$ entries of type Number and
+maps arithmetic operations to appropriate single-instruction/multiple-data
+(SIMD) concepts by intrinsic functions. The length of the vector can be
+queried by VectorizedArray::size() and its underlying number type by
+VectorizedArray::value_type.
+
+In the default case (<code>VectorizedArray<Number></code>), the vector length is
+set at compile time of the library to
+match the highest value supported by the given processor architecture.
+However, also a second optional template argument can be
+specified as <code>VectorizedArray<Number, size></code>, where <code>size</code> explicitly
+controls the  vector length within the capabilities of a particular instruction
+set. A full list of supported vector lengths is presented in the following table:
```

Comment:
This information is useful but it should really be in a more basic matrix-free tutorial (like step-37) or a documentation module.

Reply:
I would like to keep the table here since here it is the first place where we actually mess with the VectorizedArrayType